src/main/kotlin/org/rust/ide/inspections/fixes/RenameFix.kt

```diff
@@ -29,6 +30,7 @@ class RenameFix(
 
     override fun invoke(project: Project, file: PsiFile, startElement: PsiElement, endElement: PsiElement) =
         invokeLater {
-            RefactoringFactory.getInstance(project).createRename(startElement, newName).run()
+            val element = (startElement as? RsModDeclItem)?.reference?.resolve() ?: startElement
```

Comment:
Can this `resolve()` call be long-running (for example, read indices)? If yes, maybe it's worth extracting it into background thread?

Reply:
Done